### ISSUE TITLE
Unable to download a photo from memories to my phone

### ISSUE THREAD

**Issue by yurividal** [pulsejet/memories#1155]
**Describe the bug**
I cam browse and upload images to my memories just fine. But, when I click "download" on any image, nothing happens. The image is not downloaded to my phone, and no error or warning message is shown on the app.

Clicking the "share" button, shows a loading animation forever. Nothing happens.

I also noticed that if I go to the app settings, under notifications, it says "all notifications from this app are blocked" and doesn't allow me to change it to on. Not sire if this is related.

**To Reproduce**
Check video screenshot bellow:

**Screenshots**

https://github.com/pulsejet/memories/assets/8555749/1281cf7b-a0b9-4dc3-aa39-818ba102df94

**Platform:**

- OS: android 14 (Samsung OneUI 6.1)
- Memories Version: 7.2.0
- Nextcloud Version: (28.0.3)
- PHP Version: 8.2.16

- No errors on any of the docker containers
- Download works fine using memories on a browser
- Download works fine using the nextcloud official app.

Here is some adb logs i collected. (Seems to be a permission thing? maybe old target sdk?)
````
04-22 07:26:53.892 12459 13019 W DownloadManager: Path appears to be invalid: /storage/emulated/0/Download/memories/file_name.jpg
04-22 07:26:53.960 12459 21423 W DownloadManager: [105] Stop requested with status HTTP_DATA_ERROR: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
04-22 07:26:53.961 12459 21423 D DownloadManager: [105] Finished with status WAITING_TO_RETRY
04-22 07:26:53.961 12459 21423 V DownloadManager: MIME Type = null
04-22 07:26:53.971 12459 21423 I DownloadManager: Download 105 finished with status WAITING_TO_RETRY
04-22 07:26:53.971 12459 13401 I DownloadManager: Notification Clear Download 1:gallery.memories
04-22 07:26:53.973  2504  2504 D NotificationService: CancelNotificationRunnable end r = 0|com.android.providers.downloads|0|1:gallery.memories|10066, reason = 8
04-22 07:26:53.974 12459 12459 D DownloadManager: onStopJob id=105, reason=cancel() called by app, callingUid=10066 uid=10066 jobId=105
04-22 07:26:53.976  2504  7506 W JobScheduler: Job didn't exist in JobStore: f40f242 #u0a66/105 DownloadManager:com.android.providers.downloads

````

Edit:
I am running nextcloud in my internal lan, with a self-signed certificate. Could that be the issue?
````
04-22 07:36:40.978 12459 24798 W DownloadManager: [106] Stop requested with status HTTP_DATA_ERROR: java.security.cert.CertPathValidatorException: Trust anchor for certification path not found.
````

**Comment by yurividal** [comment]
edit: i've edited the report to add some adb logs i collected

**Comment by yurividal** [comment]
@pulsejet I am convinced this is being caused by the self-signed cert. Downloadmanager seems to be rejecting the untrusted CA, even though I have the CA cert installed as a trusted chain.

**Comment by pulsejet** [comment]
Yeah, that's very likely

**Comment by Termikkazan** [comment]
I still have this problem. self-signed certificate. The smartphone's built-in bootloader cannot download files. if you copy the link from the download manager and open it in the browser, it will display "{"message":"Not found (handle)"}". But I also noticed that when the files are downloaded normally into the browser version. and if you follow the link of the downloaded file again, then “{"message":"Not found (handle)"}" will also be displayed. As I understand it, when we download files, a temporary link ".../memories/api/download/xxx" is created with the ability to download the file one time. possible for one thread. And the built-in bootloader may use multithreading. Isn't that the problem?
PS. Sorry. I am not an English speaker, I use a translator.

**Comment by pulsejet** [comment]
Yes, this bug isn't resolved. It needs a major rewrite of the download logic.

**Comment by yurividal** [comment]
@pulsejet are there any updates on this?

**Comment by pulsejet** [comment]
> Yes, this bug isn't resolved. It needs a major rewrite of the download logic.

Still this unfortunately. PRs welcome.

**Comment by yurividal** [comment]
I ended up using a dns-challenge to get a valid cert for my nextcloud server. With a valid cert, everything works fine.

**Comment by MattRGX** [comment]
Hello,

any news on that issue ? 

Still can't download or share picture from either timeline or shared album on my Android 15 phone.

Memories is on my home server, debian 12, https is configured with an auto-signed cert.

Thanks in advance for your help !

**Comment by DragonDope** [comment]
> > Yes, this bug isn't resolved. It needs a major rewrite of the download logic.
> 
> Still this unfortunately. PRs welcome.

Is it possible that a pull request has already been submitted to fix this issue? #1330 
Unfortunately, it hasn't been merged yet.

**Comment by Tursu311** [comment]
> I ended up using a dns-challenge to get a valid cert for my nextcloud server. With a valid cert, everything works fine.

Done that but I still see same behaviour as with self-signed.

**Comment by AndreasFerdinand** [comment]
Hi,
unfortunately I have the same problem, with a valid certificate from let's encrypt. When I try to download photos from an shared album nothing happens. - Are there any news on this topic?
Regards
Andi
